src/vstest.console/InProcessVsTestConsoleWrapper.cs

```diff
@@ -89,9 +114,13 @@ public InProcessVsTestConsoleWrapper(ConsoleParameters consoleParameters)
         consoleParameters.PortNumber = port;
 
         // Start vstest.console.
-        // TODO: under VS we use consoleParameters.InheritEnvironmentVariables, we take that
-        // into account when starting a testhost, or clean up in the service host, and use the
-        // desired set, so all children can inherit it.
+        // Running vstest.console in process means we inherit all environment variables from the
+        // process we load the wrapper into. We do not want to alter this environment since that
+        // would mean we may interfer with the way the host process works. However, certain
+        // alterations are desired. The solution is to pass the environment variables we get via
+        // the console parameters directly to the testhost process and make sure that at least the
+        // testhost environment is predictable.
+        ProcessHelper.ExternalEnvironmentVariables = consoleParameters.EnvironmentVariables;
```

Comment:
You are not passing the option that says if we should clean up, so if user says we should not (default) and will pass just AAA="bbb", you won't be able to start testhost, because it will get just 1 env variable.

Reply:
Awesome catch, thanks ! Fixing right away 👍